File: 03-06-2024_Append_Characters_to_String_to_Make_Subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        sLength = len(s)
        tLength = len(t)
        sIndex = 0
        tIndex = 0
        
        while tIndex < tLength and sIndex < sLength:
            if s[sIndex] == t[tIndex]:
                tIndex += 1
            sIndex += 1
        
        return tLength - tIndex

File: 03-06-2024_Append_Characters_to_String_to_Make_Subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int sLength = s.length();
        int tLength = t.length();
        int sIndex = 0, tIndex = 0;

        while (tIndex < tLength && sIndex < sLength) {
            if (s.charAt(sIndex) == t.charAt(tIndex)) {
                tIndex++;
            }
            sIndex++;
        }

        return tLength - tIndex;
    }
}
